calculator.go

```diff
@@ -1,13 +1,96 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
+import (
+	"go/token"
+	"go/types"
+	"math"
+	"strconv"
+)
+
+const divideByZeroMessage = "Cannot divide by zero"
+const negativeNumber = "No negative number allowed"
+
+// ErrorString is a trivial implementation of error.
+type ErrorString struct {
+	s string
+}
+
+func (e *ErrorString) Error() string {
+	return e.s
+}
+
+func extractFirstItem(numbers []float64) (float64, []float64) {
+	return numbers[0], numbers[1:]
+}
+
 // Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+func Add(numbers ...float64) float64 {
```

Comment:
One possible problem here is that we might call Add with just one number, which seems silly, or with no numbers at all, which is even sillier! How could we ensure, via the compiler, that _at least_ two numbers are passed?